app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java

```diff
@@ -136,5 +149,72 @@ public void destroy() {
         saveEnableMobileData.dispose();
         saveImageSize.dispose();
         saveKeepScreenOn.dispose();
+        unSyncedTransmissionsExist.dispose();
+        clearAllData.dispose();
+        clearResponses.dispose();
+    }
+
+    public void deleteCollectedData() {
+        unSyncedTransmissionsExist.execute(new DefaultObserver<Boolean>(){
+            @Override
+            public void onError(Throwable e) {
+                Timber.e(e);
+                view.showDeleteCollectedData();
+            }
+
+            @Override
+            public void onNext(Boolean exist) {
+                if (exist != null && exist) {
+                    view.showDeleteCollectedDataWithPending();
+                } else {
+                    view.showDeleteCollectedData();
+                }
+            }
+        }, null);
+    }
+
+    public void deleteAllData() {
+        unSyncedTransmissionsExist.execute(new DefaultObserver<Boolean>(){
+            @Override
+            public void onError(Throwable e) {
+                Timber.e(e);
+                view.showDeleteAllData();
+            }
+
+            @Override
+            public void onNext(Boolean exist) {
```

Comment:
Same as previous comment of Boolean

Reply:
rxjava uses Boolean better use it. This how it is in all Observers